sdk/core/src/main/java/com/klaviyo/core/KLog.kt

```diff
@@ -0,0 +1,116 @@
+package com.klaviyo.core
+
+import android.os.Build
+import com.klaviyo.core.config.Config
+import com.klaviyo.core.config.Log
+import com.klaviyo.core.config.Log.Level
+import java.util.regex.Pattern
+
+open class KLog : Log {
+
+    private var _logLevel: Level? = null
+    override var logLevel: Level
+        get() {
+            if (_logLevel == null && Registry.isRegistered<Config>()) {
+                _logLevel = Registry.config.getManifestInt(LOG_LEVEL, defaultLogLevel.ordinal).let {
+                    Level.entries.getOrNull(it) ?: defaultLogLevel.also { default ->
+                        default.log(
+                            "KlaviyoLog",
+                            "Invalid log level $it detected in manifest, defaulting to $default."
+                        )
+                    }
+                }
+            }
+
+            return _logLevel ?: defaultLogLevel
+        }
+        set(value) {
```

Comment:
Since this is a public setter, you can also change log level from runtime if you import `core` from gradle